### DISCUSSION TITLE
StoreKit 2 🥳

### DISCUSSION THREAD

**Discussion by neilco** [hyochan/react-native-iap#1370] [Ideas]
There's lots of interesting changes coming in [StoreKit 2](https://developer.apple.com/wwdc21/10114). Just throwing my hat in the ring early for helping out with adding StoreKit 2 support.

**Comment by MadsonDouglas** [comment]
The Story kit 2 solves the problem of associating a user with a purchase, it would be nice if it had this feature in a next update of react-native-iap

**Reply by hyochan** [reply]
#1403

**Comment by andresesfm** [comment]
I have started looking at this. And there are two Major changes coming because of this:

1) StoreKit 2 requires ios 15 and up (Currently we are at 10)
2) We will change the model from listeners back to the method call returning 

Some other changes:
3) I will split the requests/ responses between Android and iOS since they are completely different
4) I will refactor all the Adhoc purchaseXXXIOS methods into a single one

cc @jeremybarbet since this might affect your current work with types

**Comment by andresesfm** [comment]
@jeremybarbet I think I have a good base for the native code migration. Do you think this would be a good time to merge my PR into next? Next steps are to work on the js integration of the new methods

**Reply by jeremybarbet** [reply]
To be honest, probably not, I'm still vey much in the middle of splitting my huge PR so that would make it more complicated for now. Maybe as a first step, you could base your branch off `next`, and rebase but without merging in there for now. If that's good for you 🙌 

I have one concern regarding the new StoreKit2 is the new [min iOS](https://github.com/dooboolab/react-native-iap/pull/1855/files#diff-59103f15f0c388d9eb177f2aa7673feed4fd2efd4c99ac22f7ef1c553a35a3d6R4) requirement. Wouldn't be an issue for developers using the library? `react-native` still support a [few version](https://github.com/facebook/react-native/blob/main/template/ios/Podfile#L4) old by default for iOS. I would be a bit scared that developers can't migrate to the latest version of the library, because they still need to support iOS 14, f.x.

**Reply by andresesfm** [reply]
Looks like at least 85% are on 15.0 or newer according to https://gs.statcounter.com/ios-version-market-share/all/united-states-of-america. If this is a concern, could anyone please let us know?

**Comment by andresesfm** [comment]
@jeremybarbet I have started the JS migration to the new model needed for StoreKit 2. Please note that this is not final by any stretch of the imagination. However, I think, at this point you might have some input on the following aspects:

There's no need for purchase/error listeners. There will be an update transaction listener only for iOS
I'm going to change the types to match the names of the native SDK for example `product.productId -> product.id`
We need add requestPurchase as a hook so we can store currentPurchase in the state.

**Comment by andresesfm** [comment]
I have just published `11.0.0-alpha.1` with the basic implementation of SK2. Please let me know of any issues

**Comment by andresesfm** [comment]
Published `11.0.0-alpha.6` with backwards compatiblity. No longer need to target iOS 15 in order to use the library. However, in order to use sk2 you'll need to target ios 15.0

**Comment by RocMark** [comment]
will [beginRefundRequest](https://developer.apple.com/documentation/storekit/transaction/3803220-beginrefundrequest) function add in version 11 ? 
How can i simulate refund process without StoreKit2 in testflight ?
As far as I know it can only be test in apple store am i right ?

**Reply by andresesfm** [reply]
I don't think it'll make it on 11.0.0 but I can be added later on. Please create a ticket for tracking purposes. Also please consider making a contribution to the project. Thanks

**Comment by iim-norse** [comment]
haw can i call enableStorekit2() if  i used  withIAPContext() ?

**Reply by andresesfm** [reply]
call it before creating your context

**Comment by andresesfm** [comment]
I have released `11.0.0-rc.4` which will be the last RC before moving it to stable

**Comment by andresesfm** [comment]
Released `11.0.0-rc.5` with last minute fix to `getAvailableItems`

**Comment by andresesfm** [comment]
`11.0.0` has been released
